bam/reader.go

```diff
@@ -109,90 +161,58 @@ func (br *Reader) Read() (*sam.Record, error) {
 		return nil, io.EOF
 	}
 
-	r := errReader{r: br.r}
-	bin := binaryReader{r: &r}
-
-	// Read record header data.
-	blockSize := int(bin.readInt32())
-	r.n = 0 // The blocksize field is not included in the blocksize.
-
+	var buf [4]byte
+	n, err := io.ReadFull(br.r, buf[:4])
+	if err != nil {
+		return nil, err
+	}
+	if n != 4 {
+		return nil, errors.New("bam: invalid record: short block size")
+	}
 	// br.r.Chunk() is only valid after the call the Read(), so this
```

Comment:
This comment needs to be moved with the `br.r.Begin()` call. You can fix it when you move the size read into `newBuffer`.